lib/lf/mesh/test/tp_triag_mesh_builder_tests.cc

```diff
@@ -120,4 +120,34 @@ TEST(lf_mesh_p, buildTPQuadMesh) {
   utils::PrintInfo(*mesh_p, std::cout);
 }
 
+// Test for creation of tensor product grid on torus
+TEST(lf_mesh_p, buildTorusMesh) {
+  // Enable copious output
+  hybrid2d::TPQuadMeshBuilder::output_ctrl_ = 100;
+  // Construct a tensor-product grid with 15 cells
+  std::shared_ptr<hybrid2dp::MeshFactory> mesh_factory_ptr =
+      std::make_shared<hybrid2dp::MeshFactory>(3);
+  hybrid2d::TorusMeshBuilder builder(mesh_factory_ptr);
+  // Set mesh parameters following the Builder pattern
+  builder.setBottomLeftCorner(Eigen::Vector2d{0, 0})
+      .setTopRightCorner(Eigen::Vector2d{1, 4})
+      .setNoXCells(3)
+      .setNoYCells(5);
+  auto mesh_p = builder.Build();
+
+  EXPECT_TRUE(mesh_p) << "Oops! no mesh!";
+  EXPECT_EQ(mesh_p->DimMesh(), 2) << "Mesh dimension != 2 !";
+  EXPECT_EQ(mesh_p->DimWorld(), 3) << "World dimension must be 3";
+  EXPECT_EQ(mesh_p->Size(0), 15) << "Mesh should comprise 15 cells";
+  EXPECT_EQ(mesh_p->Size(1), 30) << "Mesh should comprise 30 edges";
+  EXPECT_EQ(mesh_p->Size(2), 15) << "Mesh should have 15 vertices";
+
+  std::cout << "Checking entity indexing" << std::endl;
```

Comment:
Personally I'm not a big fan of writing to std::cout in tests because it convolutes the output considerably. Check e.g. the output that is produced by travis-ci for every commit... I think a test should only write to the command line if something goes wrong, which is exactly what you do on lines 138-143.

I know that at the moment the code is not very consistent on this, but this is something  would like to fix eventually.

Reply:
Agreed :+1:

Reply:
Agreed :+1: